aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -617,6 +647,195 @@ module aptos_framework::coin {
         let BurnCapability<CoinType> {} = burn_cap;
     }
 
+    /// Initialize fungible asset migration.
+    public entry fun initialize_fungible_asset_migration(aptos_framework: &signer) {
+        system_addresses::assert_aptos_framework(aptos_framework);
+
+        move_to(aptos_framework, FungibleAssetMigration {
+            coins_to_fungible_assets: smart_table::new(),
+            fungible_assets_to_coins: smart_table::new(),
+            enable_default_fungible_assets_in_transfers: false,
+        });
+    }
+
+    /// Create a fungible asset version of an existing Coin. This can only be called by
+    /// the signer corresponding to the address where CoinType is defined/deployed.
+    /// Icon and project uris are new fields in Fungible Asset that don't exist in Coin
+    /// and can be provided or left empty at the caller's discretion.
+    public entry fun initialize_fungible_asset<CoinType>(
+        creator: &signer,
+        maximum_supply: Option<u128>,
```

Comment:
We can check the current supply of v1 Coin (if supply is tracked) and abort if max supply is provided but < current Coin supply